build/common.ps1

```diff
@@ -200,80 +201,33 @@ Function Install-DotnetCLI {
-                if ($count -eq 1) {
-                    $expectedVersion = $line.trim()
-                }
-                $count += 1
-            }
-
-            $httpGetUrl = "https://dotnetcli.blob.core.windows.net/dotnet/Sdk/" + $expectedVersion + "/productVersion.txt"
-            $versionFile = Invoke-RestMethod -Method Get -Uri $httpGetUrl
-
-            $stringReader = New-Object -TypeName System.IO.StringReader -ArgumentList $versionFile
-            [int]$count = 0
-            while ( $line = $stringReader.ReadLine() ) {
-                if ($count -eq 1) {
-                    $specificVersion = $line.trim()
-                }
-                $count += 1
-            }
-        }
-        else {
-            $specificVersion = $Version
-        }
-
-        Trace-Log "The version of SDK should be installed is : $specificVersion"
-
-        $probeDotnetPath = Join-Path (Join-Path $cli.Root sdk)  $specificVersion
-
-        Trace-Log "Probing folder : $probeDotnetPath"
-
-        #If "-force" is specified, or folder with specific version doesn't exist, the download command will run"
-        if ($Force -or -not (Test-Path $probeDotnetPath)) {
-            $channelMainVersion = ""
-            foreach($channelPart in $cli.Channel.Split('/'))
-            {
-                if ($channelPart -match "\d+.*")
-                {
-                    $channelMainVersion = $channelPart.Split('.')[0]
-                    Break
-                }
-            }
-
-            if ([string]::IsNullOrEmpty($channelMainVersion)) {
-                Error-Log "Unable to detect channel version for dotnetinstall.ps1. The CLI install cannot be initiated." -Fatal
-            }
-
-            Trace-Log "$DotNetInstall -Channel $($channelMainVersion) -InstallDir $($cli.Root) -Version $($cli.Version) -Architecture $arch -NoPath"
-            # dotnet-install might make http requests that fail, but it handles those errors internally
-            # However, Invoke-BuildStep checks if any error happened, ever. Hence we need to run dotnet-install
-            # in a different process, to avoid treating their handled errors as build errors.
-            & powershell $DotNetInstall -Channel $channelMainVersion -InstallDir $cli.Root -Version $cli.Version -Architecture $arch -NoPath
-            if ($LASTEXITCODE -ne 0)
-            {
-                throw "dotnet-install.ps1 exited with non-zero exit code"
-            }
+        # The Quality option:
+        # Daily links are those from daily builds
+        # Signed have been post-build signed (in the case of 6.0+, pre-6.0 is signed even in daily builds)
+        # Validated have gone through CTI testing and other validation
+        # Preview are released bits that are preview versions
+        # GA are released servicing and GA builds
+        Trace-Log "$DotNetInstall -Channel $($cli.Channel) -Quality signed -InstallDir $($cli.Root) -Version $($cli.Version) -Architecture $arch -NoPath"
```

Comment:
Does `signed` imply *daily* signed builds?

Reply:
Good question!
Just asked .NET team, signed doesn't imply a daily signed build.
Michelle's reply:
It is not a daily build, no. The intent is that it is a pre-release of the currently in-flight build that is currently being considered for release. Validated are those builds that have been chosen as the release build, but before actual release day.